hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/ExtractedDatabaseMetaDataImpl.java

```diff
@@ -181,7 +181,8 @@ public Builder apply(DatabaseMetaData databaseMetaData) throws SQLException {
 			connectionCatalogName = databaseMetaData.getConnection().getCatalog();
 			// NOTE : databaseMetaData.getConnection().getSchema() would require java 1.7 as baseline
 			supportsRefCursors = StandardRefCursorSupport.supportsRefCursors( databaseMetaData );
-			supportsNamedParameters = databaseMetaData.supportsNamedParameters();
+			supportsNamedParameters = jdbcEnvironment.getDialect()
+					.supportsNamedParameters() && databaseMetaData.supportsNamedParameters();
```

Comment:
I should have mentioned also that if ``databaseMetaData.supportsNamedParameters()`` incorrectly returns false, then:
````
jdbcEnvironment.getDialect().supportsNamedParameters() && databaseMetaData.supportsNamedParameters();
````
does not allow Dialect.supportsNamedParameters() to override the incorrect value (since the result will still be ``false``).